indexing-hadoop/src/main/java/io/druid/indexer/InputRowSerde.java

```diff
@@ -49,7 +57,166 @@
 {
   private static final Logger log = new Logger(InputRowSerde.class);
 
-  public static final byte[] toBytes(final InputRow row, AggregatorFactory[] aggs, boolean reportParseExceptions)
+  private static final IndexSerdeTypeHelper STRING_HELPER = new StringIndexSerdeTypeHelper();
+  private static final IndexSerdeTypeHelper LONG_HELPER = new LongIndexSerdeTypeHelper();
+  private static final IndexSerdeTypeHelper FLOAT_HELPER = new FloatIndexSerdeTypeHelper();
+  private static final IndexSerdeTypeHelper DOUBLE_HELPER = new DoubleIndexSerdeTypeHelper();
+
+  public interface IndexSerdeTypeHelper<T>
+  {
+    ValueType getType();
+
+    byte[] serialize(Object value);
```

Comment:
`void serialize(DataOutput out, Object value)` would be better - avoid needless byte[] allocations.

Reply:
changed to  `void serialize(ByteArrayDataOutput out, Object value)`